compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -461,8 +461,11 @@ object Trees {
         else if qualifier.span.exists && qualifier.span.start > span.point then // right associative
           val realName = name.stripModuleClassSuffix.lastPart
           Span(span.start, span.start + realName.length, point)
-        else
-          Span(point, span.end, point)
+        else if span.pointMayBeIncorrect then
+          val realName = name.stripModuleClassSuffix.lastPart
+          val probablyPoint = span.end - realName.length
+          Span(probablyPoint, span.end, probablyPoint)
+        else Span(point, span.end, point)
```

Comment:
I don't even see where select gets its point; I see the span is set in `typedSelectOnTerm` for example. I haven't looked at the span computations there. But, it seems to me I can never trust the point of a huge span, and that counting back from the end is always correct in that case. Or, is there logic somewhere that overflowing point is reset to zero?

Reply:
Yes, overflowing is set to zero. The exact line is in the PR description.

Reply:
Thanks, my eyes saw it but my brain did not.

Reply:
For this particular case, the span is set in `selectorOrMatch` in `Parsers.scala` by `atSpan` method.